OpenRA.Mods.Common/Widgets/Logic/MapChooserLogic.cs

```diff
@@ -98,7 +98,7 @@ public class MapChooserLogic : ChromeLogic
 			}
 
 			var deleteMapButton = widget.Get<ButtonWidget>("DELETE_MAP_BUTTON");
-			deleteMapButton.IsDisabled = () => modData.MapCache[selectedUid].Class != MapClassification.User;
+			deleteMapButton.IsDisabled = () => currentTab != MapClassification.User;
```

Comment:
The delete button was querying `mapCache` every frame, defeating the whole purpose of the queue. 

All it wanted to know is if the map chooser tab was changed. As as `currentTab` is already updated on a switch, there's no need to query `mapChache`